src/v/http/utils.cc

```diff
@@ -49,4 +51,13 @@ ss::sstring uri_encode(std::string_view input, uri_encode_slash encode_slash) {
     return result;
 }
 
+iobuf form_encode_data(
+  const absl::flat_hash_map<ss::sstring, ss::sstring>& data) {
+    std::vector<ss::sstring> pairs;
+    for (const auto& [k, v] : data) {
+        pairs.emplace_back(fmt::format("{}={}", k, v));
+    }
+    return iobuf::from(boost::algorithm::join(pairs, "&"));
+}
```

Comment:
Do we need to worry about escaping at all? It always feels sketchy to have a generic method in a utils target that doesn't handle edge cases like that.

Reply:
good point, added percent encoding. also added a test